### PR TITLE
DEP: bump required versions of dependecies

### PR DESCRIPTION
**Pull request by martinfleis**
Bumping required versions of dependencies following the rules defined in #1457 and removed now obsolete compatibility code.

### CODE REVIEW

**Comment by martinfleis** [comment]
I think that we're hitting #1697 in environments coming from `defaults`. They're latest fiona is quite old and seem to conflict with a newer pyproj...

**Comment by martinfleis** [comment]
And why are we getting `package python-3.7.9-4_73_pypy requires python_abi 3.7 *_pypy37_pp73, but none of the providers can be installed` in the minimal env, the I truly don't know. I'll fire up my Linux box to try these locally.

**Comment by martinfleis** [comment]
I think we should be fine now. A few observation as it was not as easy as I thought.

- when using defaults, `pyproj>3` does not work together with Fiona, so it only resolves with pyproj2.6 (good to know because then next release should bump pyproj to v3). When installed via pip we bump into #1821 as a result of conflicting PROJ databases.
- some of pandas 1.0.x contain a regression causing GeoDataFrame to be a DataFrame. 1.0.5 has that one fixed but we should keep in mind that on 1.0.0 and 1.0.1 (maybe some more), GeoPandas does not work.

File: geopandas/tests/test_array.py [hunk 1]
```diff
@@ -922,7 +909,8 @@ def test_estimate_utm_crs__geographic(self):
                 self.landmarks.estimate_utm_crs()
         else:
             assert self.landmarks.estimate_utm_crs() == CRS("EPSG:32618")
-            assert self.landmarks.estimate_utm_crs("NAD83") == CRS("EPSG:26918")
+            if compat.PYPROJ_GE_32:  # result is unstable in older pyproj
+                assert self.landmarks.estimate_utm_crs("NAD83") == CRS("EPSG:26918")
```
Comment (jorisvandenbossche) [inline geopandas/tests/test_array.py:913]: Didn't we test with such older version before, or is it triggered due to the combo with a certain fiona install?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks a lot! Looks good, one more thing: can you also update the documentation? (https://geopandas.org/en/latest/getting_started/install.html#dependencies)

File: geopandas/tests/test_array.py (hunk 1 above)
Comment (martinfleis) [reply geopandas/tests/test_array.py:913]: No. We tested with 2.2 which skipped the whole test and latest. I have added 3.1 which causes the issue #2045

**Comment by martinfleis** [comment]
>  can you also update the documentation? 

Forgot about that one :). Fixed.

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**